wikimedia/wikipedia-ios
# Final API tweaks

**Phabricator:** 
https://phabricator.wikimedia.org/T348278
https://phabricator.wikimedia.org/T345847
Reverting https://phabricator.wikimedia.org/T345852

### Notes
These are some additional tweaks needed against the FR Tech endpoints, as well as tweaks needed against the new campaign configs:

- Remove injection scripts
- Remove Payments API Key from calls to FR Tech endpoints
- Add banner ID (when coming from campaign modal...`utm_source` value from campaign) to `submitPayment` API call.
- Add app version to `submitPayment` API call
- Updated config mock with latest from test wiki campaign [config](https://test.wikipedia.org/wiki/MediaWiki:AppsCampaignConfig.json). Add Italian campaign ID as an ID we're targeting to receive the new donor experience.

I'm going to mark this as Draft in case there are any final tweaks from FR Tech. This is dependent on https://github.com/wikimedia/wikipedia-ios/pull/4649.

### Test Steps
1. Make these local changes to mock endpoints on staging - https://github.com/wikimedia/wikipedia-ios/pull/4645/commits/6eaaa0b03bda2a80749b744d6cd0929147c463dd
2. Change device region to Italy
3. Run app on Staging. Background app, foreground, pull to refresh on Explore feed.
4. Go to an article on IT Wikipedia. Confirm you see campaign modal in Italian. Tap "Donate". Tap "Apple Pay".
5. Line break [here](https://github.com/wikimedia/wikipedia-ios/compare/new-fundraising-banner...fr-tech-tweaks?expand=1#diff-3e91028af2c7e6e2f00f3ba9cd5d4b94a7eb74c728848a4bdf3ebf5df3f95becR176).
6. On donate form, go through donation flow until app tries to submit. Check at line break and be sure banner ID and app version is populated.
